FreeRTOS/FreeRTOS-Kernel
# Add IAR RISC-V 32 Embedded Extension Support

Description
-----------
Adds RISC-V 32-bit Embedded extension support. The Embedded or "E" [extension](https://en.wikipedia.org/wiki/RISC-V#ISA_base_and_extensions) proposes a reduced register set (down to 16 from 32) to prioritize interrupt and contact switching speed [[1]](https://github.com/riscv-non-isa/riscv-eabi-spec/blob/master/EABI.adoc).

Test Steps
-----------
Flashed to SiFive HiFive1 Rev B with JTAG header as covered in [this page](https://www.freertos.org/RTOS-RISC-V-FreedomStudio-IAR-HiFive-RevB.html#build_instructions_iar). Note - I did need to have the HiFive board plugged in over USB micro as well.

Blinky demo executed without issue.

Related Issue
-----------
None


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
